### PR TITLE
feat: redirect built-in APIs to deno.land/api and /x modules to /x

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: routes/doc.tsx [hunk 1]
```diff
@@ -150,6 +150,15 @@ export const pathGetHead = async <R extends DocRoutes>(
   ctx: RouterContext<R>,
 ) => {
   let { proto, host, item, path } = ctx.params;
+  if (proto === "deno") {
+    ctx.response.status = Status.MovedPermanently;
+    if (host === "unstable") {
+      ctx.response.redirect("https://deno.land/api");
+    } else {
+      ctx.response.redirect("https://deno.land/api?unstable");
```
Comment (ry) [inline routes/doc.tsx:158]: inverted?

File: routes/doc.tsx [hunk 2]
```diff
@@ -150,6 +150,15 @@ export const pathGetHead = async <R extends DocRoutes>(
...
+    }
+    return;
```
Comment (ry) [inline routes/doc.tsx:160]: why `return;`?

File: routes/doc.tsx (hunk 1 above)
Comment (crowlKats) [reply routes/doc.tsx]: fixed

File: routes/doc.tsx (hunk 2 above)
Comment (crowlKats) [reply routes/doc.tsx:160]: because we dont want any of the code in the other branch to be run if we are redirecting

Comment (kitsonk) [reply routes/doc.tsx:174]: Just for unambiguity, I would combine the return with the previous line.  It isn't specifically required, but it should raise less questions for those reading the code.

**Review by kitsonk — COMMENTED** [review]
I think we would want/need to move the tests to something like `https://doc.deno.land/https://esm.sh/v96/@firebase/firestore@3.5.0/dist/index.d.ts`

**Pull request closed without merging**